client/workflow.go

```diff
@@ -0,0 +1,310 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+
+	pb "github.com/dapr/dapr/pkg/proto/runtime/v1"
+)
+
+type StartWorkflowRequest struct {
+	InstanceID        string // Optional instance identifier
+	WorkflowComponent string
+	WorkflowName      string
+	Options           map[string]string // Optional metadata
+	Input             []byte            // Optional input
+	// TODO: support data serialization
+}
+
+type StartWorkflowResponse struct {
+	InstanceID string
+}
+
+type GetWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+}
+
+type GetWorkflowResponse struct {
+	InstanceID    string
+	WorkflowName  string
+	CreatedAt     time.Time
+	LastUpdatedAt time.Time
+	RuntimeStatus string
+	Properties    map[string]string
+}
+
+type PurgeWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+}
+
+type TerminateWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+}
+
+type PauseWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+}
+
+type ResumeWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+}
+
+type RaiseEventWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+	EventName         string
+	EventData         []byte // Optional data
+}
+
+// StartWorkflowAlpha1 starts a workflow instance using the alpha1 spec.
+func (c *GRPCClient) StartWorkflowAlpha1(ctx context.Context, req *StartWorkflowRequest) (*StartWorkflowResponse, error) {
+	if req.InstanceID == "" {
+		req.InstanceID = uuid.New().String()
+	}
+	if req.WorkflowComponent == "" || req.WorkflowName == "" {
```

Comment:
Can we separate this if check into 2 separate checks with separate errors? For a user, if they provide one or the other, they won't know immediately whether or not they didn't supply the WorkflowComponent or the WorkflowName. This comment would apply in all the functions below as well.